nemo/lightning/nemo_logger.py

```diff
@@ -220,7 +220,7 @@ def _setup_trainer_model_checkpoint(self, trainer, log_dir, ckpt=None):
                 if callback.dirpath is None:
                     callback.dirpath = Path(log_dir / "checkpoints")
                 if callback.filename is None:
-                    callback.filename = f"{self.name}--{{{callback.monitor}:.4f}}-{{epoch}}"
+                    callback.filename = f"{self.name}--{{{callback.monitor}:.4f}}-{{epoch}}-{{consumed_samples}}"
```

Comment:
do any corresponding changes need to be made for looking up checkpoints to load from, especially in the auto-resume case?

Reply:
good point. Looking at the 'resume_if_exists' path in AutoResume, the [logic](https://github.com/NVIDIA/NeMo/blob/maanug/default-ckpt-name/nemo/lightning/resume.py#L185-L246) for selecting the -last checkpoint should not be affected.

Reply:
Some CI tests that hardcode a checkpoint path to resume from will be affected though.